### PR TITLE
Updated nosharp.py

### PR DESCRIPTION
**Pull request by kmsiapps**
- Added options to turn on/off
- Now puts player on spectator team instead of kicking

### CODE REVIEW

**Comment by swalladge** [comment]
The only problem I can see is that players with invalid names can sit in the spectators filling up the connection (are players' slots filled by spectators?) and still can't be kicked

**Comment by noway** [comment]
Yep. Autokick in 10 secs would be nice

**Comment by NotAFile** [comment]
maybe an autokick after, say, 15 seconds would be a good idea

**Comment by kmsiapps** [comment]
Added autokick.

File: feature_server/config/scripts/nosharp.py [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
-                    
-            return connection.on_spawn(self, pos)
+                    self.send_chat('%% Your name starts with #. Will be kicked soon.')
+                    self.autokick_call = reactor.callLater(autokick_duration, self.autokick)
+            else:        
+                return connection.on_spawn(self, pos)
+                
+        def autokick(self):
+            self.autokick_call = None
+            self.kick('Invalid name')
```
Comment (NotAFile) [inline feature_server/config/scripts/nosharp.py]: have you checked if this reason works in both OS and AoS? Last I heard, arbitrary reasons were not allowed.

Comment (yvt) [inline feature_server/config/scripts/nosharp.py]: Arbitrary kick reasons aren't supported by neither of AoS and OS. I reckon it's only displayed on other players' chat log window and the server log.

Comment (kmsiapps) [inline feature_server/config/scripts/nosharp.py]: @NotAFile It is shown in-game to other players.

**Pull request merged**